src/librustc/hir/lowering.rs

```diff
@@ -1493,6 +1492,7 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
             opaque_ty_def_index,
             &hir_bounds,
         );
+        let lifetime_defs = self.arena.alloc_from_iter(lifetime_defs.into_iter());
```

Comment:
It's a bit odd that one of the results from `lifetimes_from_impl_trait_bounds` is on the arena and the other is owned.